js/filters.js

```diff
@@ -115,9 +145,27 @@ function TypeaheadFilter(elm) {
   var key = this.$body.data('dataset');
   var dataset = typeahead.datasets[key];
   this.typeaheadFilter = new typeaheadFilter.TypeaheadFilter(this.$body, dataset);
```

Comment:
This is basically a historical accident. The `TypeaheadFilter` class was written well before we added the `Filter` and `FilterSet` abstractions--so rather than rewrite the earlier code, this class just wraps a nested legacy typeahead. It doesn't look like the nested object is ever used on its own (although I guess it could be at some point), so I think it would make sense to merge the two classes. But I would defer this to a separate patch, since this one's focused on adding filter tags.